Mathlib/Data/Fin/Basic.lean

```diff
@@ -475,6 +475,14 @@ attribute [simp] castSucc_inj
 lemma castLE_injective (hmn : m ≤ n) : Injective (castLE hmn) :=
   fun _ _ hab ↦ Fin.ext (congr_arg val hab :)
 
+@[simp]
+lemma castLE_natCast {m n : ℕ} [NeZero m] (h : m ≤ n) (a : ℕ) :
+    letI : NeZero n := ⟨Nat.pos_iff_ne_zero.mp (lt_of_lt_of_le (Nat.pos_of_neZero m) h)⟩
+    Fin.castLE h a = (a % m : ℕ) := by
```

Comment:
Given that this uses a locally defined instance, it's not clear to me that `simp` will actually fire on it?

Reply:
Yes, since the instance is only used on the rhs of the statement